webplatformcompat/models.py

```diff
@@ -85,12 +96,39 @@ class Feature(MPTTModel):
         null=True, blank=True, related_name='children')
     sections = SortedManyToManyField(
         'Section', related_name='features', blank=True)
-    objects = CachingManager()
+    objects = CachingTreeManager()
     # history = HistoricalFeatureRecords()  # Registered below
 
     def __str__(self):
         return self.slug
 
+    def set_children_order(self, children):
+        """Set the child features in the given order.
+
+        django-mptt doesn't have a function to do this, and uses direct SQL
+        to change the tree, so a lot of reloading is required to get it right.
+        """
+        # Verify that all children are present
+        current_children = list(self.get_children())
+        current_set = set([child.pk for child in current_children])
+        new_set = set([child.pk for child in children])
+        assert current_set == new_set, "Can not add/remove child features."
+
+        # Set order, refreshing as we go
+        prev_child = None
+        moved = False
+        for pos, next_child in enumerate(children):
+            if current_children[pos].pk != next_child.pk:
+                if moved:
+                    next_child.refresh_from_db()
+                if prev_child is None:
+                    next_child.move_to(self)
```

Comment:
Updated to ``first-child``, which is good, because the default was ``last-child``.